### PR TITLE
🐛 properly wait until progress bar is closed

### PR DESCRIPTION
**Pull request by chris-rock**
The terminal cursor was not properly recovered because we have not waited until the go routine is properly completed

### CODE REVIEW

File: cli/progress/progress.go [hunk 1]
```diff
@@ -106,7 +111,10 @@ func (p *progressbar) OnProgress(current int, total int) {
 }
 
 func (p *progressbar) Close() {
+	p.lock.Lock()
 	p.Data.complete = true
```
Comment (jaym) [inline cli/progress/progress.go:118]: if Close and Update are in different threads, you need to take a lock before writing and reading.

**Review by jaym — APPROVED** [review]

File: cli/progress/progress.go (hunk 1 above)
Comment (chris-rock) [reply cli/progress/progress.go:118]: Anything for me todo or is that as it should be?

Comment (jaym) [reply cli/progress/progress.go:118]: Sorry, got the function name wrong. The uses of `p.Data.complete` in `Open`  need to take a lock before looking a `p.Data.complete`

Comment (chris-rock) [reply cli/progress/progress.go:118]: great catch @jaym

**Comment by chris-rock** [comment]
@jaym I updated the PR, can you double-check that I did this right?

**Comment by jaym** [comment]
lgtm

**Pull request merged**